### PR TITLE
[Merged by Bors] - chore: `Finsupp.coe_finset_sum` simp

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanCamCombi

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Finsupp.lean [hunk 1]
```diff
@@ -249,12 +249,12 @@ theorem sum_apply [Zero M] [AddCommMonoid N] {f : α →₀ M} {g : α → M →
   finset_sum_apply _ _ _
 
 -- Porting note: inserted ⇑ on the rhs
-theorem coe_finset_sum [AddCommMonoid N] (S : Finset ι) (f : ι → α →₀ N) :
+@[simp, norm_cast] theorem coe_finset_sum [AddCommMonoid N] (S : Finset ι) (f : ι → α →₀ N) :
```
Comment (b-mehta) [inline Mathlib/Algebra/BigOperators/Finsupp.lean:252]: Given that the only downstream change from this is to require _removing_ the simp attribute, this change isn't clearly positive. Can you justify why the regression is acceptable?

Comment (YaelDillies) [reply Mathlib/Algebra/BigOperators/Finsupp.lean:252]: Actually, there is no regression anymore. The proof was just suboptimally written, with `← IH` going against the confluence

File: Mathlib/Data/Nat/Factorization/Defs.lean [hunk 2]
```diff
@@ -172,7 +172,7 @@ theorem factorization_prod {α : Type*} {S : Finset α} {g : α → ℕ} (hS : 
     · simp
     · intro x T hxS hTS hxT IH
       have hT : T.prod g ≠ 0 := prod_ne_zero_iff.mpr fun x hx => hS x (hTS hx)
-      simp [prod_insert hxT, sum_insert hxT, ← IH, factorization_mul (hS x hxS) hT]
+      simp [prod_insert hxT, sum_insert hxT, ← IH, factorization_mul (hS x hxS) hT, -coe_finset_sum]
```
Comment (YaelDillies) [inline Mathlib/Data/Nat/Factorization/Defs.lean:175]:
```suggestion
      simp [prod_insert hxT, sum_insert hxT, IH, factorization_mul (hS x hxS) hT]
```

**Comment by b-mehta** [comment]
Please follow the PR title conventions. You can find them here: https://leanprover-community.github.io/contribute/commit.html

bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**